methods/array_ops/src/pg_gp/array_ops.sql_in

```diff
@@ -602,11 +602,22 @@ m4_ifdef(`__HAS_FUNCTION_PROPERTIES__', `NO SQL', `');
  *        in later GPDB and Postgres versions.
  */
 DROP AGGREGATE IF EXISTS MADLIB_SCHEMA.array_agg(anyelement) CASCADE;
-CREATE AGGREGATE MADLIB_SCHEMA.array_agg( anyelement) (
-   SFUNC     = array_append,
-   STYPE     = anyarray
-   m4_ifdef( `__POSTGRESQL__', `', `, PREFUNC   = array_cat')
-);
+DO $$
+BEGIN
+  --
+  -- array_agg() compatibility function is only needed PostgreSQL 9.4 or earlier
+  -- because 9.5 or later has pg_catalog.array_agg(anyarray).
+  --
+  IF regexp_replace(version(), '.*PostgreSQL (\d+\.\d+).*', '\1')::float8 < 9.5 THEN
```

Comment:
If we can drop this function at all, it would make things simpler. Does the latest GPDB have array_agg(anyarray) function? I'm not familiar with GPDB.

I'm not sure MADlib could work with 9.4 without this function, but at least, we need to drop this function for 9.5 or later because the function signature causes a conflict error with the built-in one.

BTW, is there any MADlib docs mentioning "current supported platform"? I could not find it...